builtin/diff.c

```diff
@@ -437,6 +437,9 @@ int cmd_diff(int argc, const char **argv, const char *prefix)
 
 	prefix = setup_git_directory_gently(&nongit);
 
+	prepare_repo_settings(the_repository);
```

Comment:
Since you include my `Co-authored-by:` on the commit, please also include my `Signed-off-by:` (before yours, since you are the last to "touch" it).